### PR TITLE
Small refactor

### PR DESCRIPTION
**Pull request by rocking5566**
This PR include
1. Remove kIsFp8 in BlockFmhaPipelineProblem
2. Use alias for fp8 warp gemm

### CODE REVIEW

File: include/ck_tile/ops/fmha/pipeline/block_fmha_pipeline_qx_ks_vs_custom_policy.hpp [hunk 1]
```diff
@@ -97,16 +97,15 @@ struct BlockFmhaPipelineQXCustomPolicy</* QLoadOnce = */ true>
             {
                 return WarpGemmMfmaBf16Bf16F32M16N16K32SwizzleBTransposedCDistribution{};
             }
-            else if constexpr(Problem::kIsFp8)
+            else if constexpr(std::is_same_v<typename Problem::QDataType, fp8_t> &&
+                              std::is_same_v<typename Problem::KDataType, fp8_t> &&
+                              std::is_same_v<typename Problem::SaccDataType, float>)
```
Comment (poyenc) [inline include/ck_tile/ops/fmha/pipeline/block_fmha_pipeline_qx_ks_vs_custom_policy.hpp:102]: don't we have to check `VDataType` here?

Comment (rocking5566) [reply include/ck_tile/ops/fmha/pipeline/block_fmha_pipeline_qx_ks_vs_custom_policy.hpp:102]: This is GetQKBlockGemm(), it is nothing to do with V

**Comment by poyenc** [comment]
LGTM

**Review by poyenc — APPROVED** [review]

**Review by carlushuang — APPROVED** [review]
LGTM

**Pull request merged**